vector/union.go

```diff
@@ -6,40 +6,45 @@ import (
 )
 
 type Union struct {
-	mem
 	Typ    *zed.TypeUnion
-	Tags   []int32
+	Tags   []uint32
+	TagMap TagMap
 	Values []Any
+	Nulls  *Bool
 }
 
 var _ Any = (*Union)(nil)
 
-func NewUnion(typ *zed.TypeUnion) *Union {
-	return &Union{Typ: typ, Values: make([]Any, len(typ.Types))}
+func NewUnion(typ *zed.TypeUnion, tags []uint32, vals []Any, nulls *Bool) *Union {
+	return &Union{
+		Typ:    typ,
+		Tags:   tags,
+		TagMap: *NewTagMap(tags, vals),
+		Values: vals,
```

Comment:
Don't forget nulls!

In constructors, I think a composite literal with a limited number of fields is often clearer without fields names. And if you omit them, you'll get a compilation error if you forget a field like this.
```suggestion
	return &Union{typ, tags, *NewTagMap(tags, vals), vals, nulls}
```